addons/web/static/src/core/navigable_popover/components/popover.js

```diff
@@ -0,0 +1,126 @@
+/** @odoo-module **/
+
+import {
+    Component,
+    xml,
+    useRef,
+    useChildSubEnv,
+    onWillUpdateProps,
+    onMounted,
+    onWillDestroy,
+    onRendered,
+} from "@odoo/owl";
+import { usePopover } from "@web/core/popover/popover_hook";
+import { useForwardRefToParent } from "@web/core/utils/hooks";
+
+/**
+ * This private custom popover will instantiate a slot given through his
+ * props.
+ *
+ * A custom environment is also given to the slot, this allow a more natural
+ * use of "useSubEnv" in the parent that uses the Popover component.
+ *
+ * The props "renderRef" is used so the slots can be properly re-rendered
+ * when the parent component has re-rendered.
+ */
+class PopoverScaffold extends Component {
+    static template = xml`
+        <div t-ref="containerRef" class="o_popover_container"  t-att-class="props.containerClass">
+            <t t-slot="default"/>
+        </div>
+    `;
+    static props = {
+        slots: { type: Object, optional: true },
+        close: { type: Function },
+        onOpened: { type: Function, optional: true },
+        onClosed: { type: Function, optional: true },
+        env: { type: Object, optional: true },
+        containerClass: { type: String, optional: true },
+        containerRef: { type: Function, optional: true },
+        renderRef: { type: Object, optional: true },
+    };
+
+    setup() {
+        this.isOpen = false;
+        this.containerRef = useRef("containerRef");
+        useForwardRefToParent("containerRef");
+
+        useChildSubEnv(this.props.env);
+
+        onMounted(() => {
+            this.props.onOpened?.();
+        });
+
+        onWillDestroy(() => {
+            this.props.onClosed?.();
+        });
+
+        if (this.props.renderRef) {
+            this.props.renderRef.render = () => this.render();
+        }
```

Comment:
looks hacky

Reply:
This is indeed quite hacky but unfortunately, because the content of the popover is given via slots, via props, it does not properly re-render when the parent component does. Without this hack a popover with dynamic content (like filterable items) does not work properly.
It might be an Owl problem and I've already created an issue on the Owl repo but in the meanwhile I think that this hack is small enough to be acceptable. But I'm of course open for a cleaner solution.